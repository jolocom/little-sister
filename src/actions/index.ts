import * as accountActions from 'src/actions/account/'
import * as registrationActions from 'src/actions/registration/'
import * as navigationActions from 'src/actions/navigation/'
import * as genericActions from 'src/actions/generic/'
import * as ssoActions from 'src/actions/sso/'

export {
  accountActions,
  registrationActions,
  navigationActions,
  genericActions,
  ssoActions,
}
