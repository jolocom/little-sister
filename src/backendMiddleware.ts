import { JolocomLib } from 'jolocom-lib'
import { IdentityWallet } from 'jolocom-lib/js/identityWallet/identityWallet'
import { EncryptionLib, EncryptionLibInterface } from 'src/lib/crypto'
import { Storage } from 'src/lib/storage/storage'
import { KeyChain, KeyChainInterface } from 'src/lib/keychain'
import { ConnectionOptions } from 'typeorm/browser'
import { SoftwareKeyProvider } from 'jolocom-lib/js/vaultedKeyProvider/softwareProvider'
import { IRegistry } from 'jolocom-lib/js/registries/types'
import { createJolocomRegistry } from 'jolocom-lib/js/registries/jolocomRegistry'
import { IpfsCustomConnector } from './lib/ipfs'
import {jolocomContractsAdapter} from 'jolocom-lib/js/contracts/contractsAdapter'
import {jolocomEthereumResolver} from 'jolocom-lib/js/ethereum/ethereum'
import {jolocomContractsGateway} from 'jolocom-lib/js/contracts/contractsGateway'

export class BackendMiddleware {
  identityWallet!: IdentityWallet
  storageLib: Storage
  encryptionLib: EncryptionLibInterface
  keyChainLib: KeyChainInterface
  registry: IRegistry

  constructor(config: {
    fuelingEndpoint: string
    typeOrmConfig: ConnectionOptions
  }) {
    this.storageLib = new Storage(config.typeOrmConfig)
    this.encryptionLib = new EncryptionLib()
    this.keyChainLib = new KeyChain()
    this.registry = createJolocomRegistry({
      ipfsConnector: new IpfsCustomConnector({
        host: 'ipfs.jolocom.com',
        port: 443,
        protocol: 'https',
      }),
      ethereumConnector: jolocomEthereumResolver,
      contracts: {
        adapter: jolocomContractsAdapter,
        gateway: jolocomContractsGateway,
      },
    })
  }

  async setIdentityWallet(
    userVault: SoftwareKeyProvider,
    pass: string,
  ): Promise<void> {
    const { jolocomIdentityKey } = JolocomLib.KeyTypes
    this.identityWallet = await this.registry.authenticate(userVault, {
      encryptionPass: pass,
      derivationPath: jolocomIdentityKey,
    })
  }
}
