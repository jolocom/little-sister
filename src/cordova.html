<!DOCTYPE html>

<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport"
          content="width=device-width, initial-scale=1, user-scalable=no">

    <title>Jolocom</title>

    <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,300,600"
          rel="stylesheet"
          type="text/css" />
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <style type="text/css">
      * {
        margin: 0;
        padding: 0;
      }
      html, body {
        height: 100%;
      }
      body {
        font-family: Roboto;
      }
      #app {
        width: 100%;
        height: 100%;
      }
    </style>

    <meta http-equiv="Content-Security-Policy" content="default-src https://codepush.azurewebsites.net https://proxy.jolocom.net https://identity.jolocom.com https://fonts.gstatic.com 'self' data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src https://fonts.googleapies.com 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline'; media-src *; ; img-src 'self' data: content:;">
    <meta name="format-detection" content="telephone=no">
    <meta name="msapplication-tap-highlight" content="no">
    <meta name="viewport" content="user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width">
  </head>
  <body>
    <div id="app"></div>
    <script src="cordova.js"></script>
    <script>
      window.addEventListener('load', function() {
        document.body.addEventListener('touchmove', function(e) {
          e.preventDefault()
        }, false)
      }, false)

      document.addEventListener('resume', function () {
        window.codePush.sync()
      })
    </script>
    <script src="js/bundle.js"></script>
  </body>
</html>
