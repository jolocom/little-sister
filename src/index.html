<!DOCTYPE html>

<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport"
          content="width=device-width, initial-scale=1, user-scalable=no">

    <title>Little Sister</title>

    <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,300,600"
          rel="stylesheet"
          type="text/css" />
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <style type="text/css">
      * {
        margin: 0;
        padding: 0;
      }
      html, body {
        height: 100%;
      }
      body {
        font-family: Roboto;
      }
      #app {
        width: 100%;
        height: 100%;
      }
    </style>

    <!-- <meta http-equiv="Content-Security-Policy" content="default-src * data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src * 'unsafe-inline'; media-src *; img-src * data:"> -->
    <meta name="format-detection" content="telephone=no">
    <meta name="msapplication-tap-highlight" content="no">
    <meta name="viewport" content="user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width">
  </head>
  <body>
    <div id="app"></div>
    <script src="https://solid.github.io/releases/rdflib.js/rdflib-0.7.0.min.js"></script>
    <script src="https://d3js.org/d3.v4.min.js"></script>
    <script src="js/bundle.js"></script>
  </body>
</html>
