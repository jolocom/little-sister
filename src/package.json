{
  "name": "src",
  "dependencies": {
    "squel": "^5.12.1"
  }
}
