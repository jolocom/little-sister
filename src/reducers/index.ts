import { combineReducers } from 'redux'
import { accountReducer, AccountState } from 'src/reducers/account/'
import {
  registrationReducer,
  RegistrationState,
} from 'src/reducers/registration/'
import { navigationReducer } from 'src/reducers/navigation/'
import { NavigationState } from 'react-navigation'
import { ssoReducer, SsoState } from 'src/reducers/sso/'

export const rootReducer = combineReducers({
  account: accountReducer,
  registration: registrationReducer,
  navigation: navigationReducer,
  sso: ssoReducer,
})

export interface RootState {
  readonly account: AccountState
  readonly registration: RegistrationState
  readonly navigation: NavigationState
  readonly sso: SsoState
}
