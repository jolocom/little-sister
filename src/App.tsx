import React from 'react'
import { Provider } from 'react-redux'
import { Navigator } from 'src/NavigatorContainer'
import { store } from 'src/store'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import SplashScreen from 'react-native-splash-screen'

const { ThemeProvider } = require('react-native-material-ui')
const assign = require('object.assign/implementation')

Object.assign = assign

// tslint:disable-next-line: no-default-export
export default class App extends React.Component {
  componentDidMount() {
    SplashScreen.hide()
  }

  render() {
    return (
      <ThemeProvider uiTheme={JolocomTheme}>
        <Provider store={store}>
          <Navigator dispatch={store.dispatch} />
        </Provider>
      </ThemeProvider>
    )
  }
}
