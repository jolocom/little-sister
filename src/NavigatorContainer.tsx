import * as React from 'react'
import { addNavigationHelpers } from 'react-navigation'
import { connect } from 'react-redux'
import { BackHandler } from 'react-native'
import { AnyAction } from 'redux'
import { Routes } from 'src/routes'
import { RootState } from 'src/reducers/'
import { navigationActions } from 'src/actions/'

const { createReduxBoundAddListener } = require('react-navigation-redux-helpers')

interface ConnectProps {
  navigation: any;
  goBack: () => void;
}

interface OwnProps {
  dispatch: (action: AnyAction) => void;
}

interface Props extends ConnectProps, OwnProps {}

class Navigator extends React.Component<Props> {
  componentWillMount() {
    BackHandler.addEventListener('hardwareBackPress', this.navigateBack)
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.navigateBack)
  }

  private navigateBack = () => {
    this.props.goBack()
    return true
  }

  render() {
    console.log(this.props)
    return (
      <Routes navigation={addNavigationHelpers({
        dispatch: this.props.dispatch,
        state: this.props.navigation,
        addListener: createReduxBoundAddListener('root')
      })}/>
    )
  }
}

const mapStateToProps = (state: RootState) => {
  return {
    navigation: state.navigation
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    goBack: () => dispatch(navigationActions.goBack())
  }
}

export const NavigatorContainer = connect(mapStateToProps, mapDispatchToProps)(Navigator)
