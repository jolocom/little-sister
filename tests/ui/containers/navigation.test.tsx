import React from 'react'
import { NavigatorContainer } from 'src/NavigatorContainer'
import { shallow } from 'enzyme'
import { RootState } from 'src/reducers/'
const {
  createReactNavigationReduxMiddleware,
} = require('react-navigation-redux-helpers')

describe('NavigatorContainer', () => {
  const COMMON_PROPS = {
    parseJWT: jest.fn(),
    goBack: jest.fn(),
    navigation: {
      index: 0,
      routes: [{ routeName: 'Home' }],
      navigate: jest.fn(),
    },
    openScanner: jest.fn(),
    handleDeepLink: jest.fn(),
    checkIfAccountExists: jest.fn(),
    dispatch: jest.fn(),
  }

  it('mounts correctly and matches snapshot', () => {
    createReactNavigationReduxMiddleware(
      'root',
      (state: RootState) => state.navigation,
    )
    const navigation = {
      index: 0,
      routes: [{ routeName: 'Home' }],
      navigate: jest.fn(),
    }
    const homeScreenProps = { ...COMMON_PROPS, navigation }

    const rendered = shallow(<NavigatorContainer {...homeScreenProps} />)
    expect(rendered).toMatchSnapshot()
  })

  it('correctly hides bottom action bar', () => {
    createReactNavigationReduxMiddleware(
      'root',
      (state: RootState) => state.navigation,
    )
    const navigation = {
      index: 0,
      routes: [{ routeName: 'Landing' }],
      navigate: jest.fn(),
    }
    const landingPageProps = { ...COMMON_PROPS, navigation }

    const rendered = shallow(<NavigatorContainer {...landingPageProps} />)
    expect(rendered).toMatchSnapshot()
  })
})
