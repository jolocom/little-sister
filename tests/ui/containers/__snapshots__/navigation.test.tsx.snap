// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigatorContainer correctly hides bottom action bar 1`] = `
Array [
  <StatusBar
    animated={false}
    barStyle="light-content"
    key="0"
    showHideTransition="fade"
  />,
  <NavigationContainer
    key="1"
    navigation={
      Object {
        "addListener": [Function],
        "dispatch": [MockFunction],
        "getParam": [Function],
        "goBack": [Function],
        "navigate": [Function],
        "pop": [Function],
        "popToTop": [Function],
        "push": [Function],
        "replace": [Function],
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "navigate": [MockFunction],
          "routes": Array [
            Object {
              "routeName": "Landing",
            },
          ],
        },
      }
    }
  />,
  "",
]
`;

exports[`NavigatorContainer mounts correctly and matches snapshot 1`] = `
Array [
  <StatusBar
    animated={false}
    barStyle="light-content"
    key="0"
    showHideTransition="fade"
  />,
  <NavigationContainer
    key="1"
    navigation={
      Object {
        "addListener": [Function],
        "dispatch": [MockFunction],
        "getParam": [Function],
        "goBack": [Function],
        "navigate": [Function],
        "pop": [Function],
        "popToTop": [Function],
        "push": [Function],
        "replace": [Function],
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "navigate": [MockFunction],
          "routes": Array [
            Object {
              "routeName": "Home",
            },
          ],
        },
      }
    }
  />,
  <BottomActionBar
    key="2"
    openScanner={[MockFunction]}
  />,
]
`;
